torch/testing/_internal/distributed/distributed_test.py

```diff
@@ -4936,6 +4936,54 @@ def _test_post_localSGD_optimizer_parity(self, create_averager, grad_is_view):
         def _create_periodic_model_averager(self):
             return averagers.PeriodicModelAverager(period=4, warmup_steps=10)
 
+        def _test_post_localSGD_optimizer_step_reload(self, create_averager):
+            learning_rate = 0.03
+            chkpt_file = os.path.join(os.environ["TEMP_DIR"], "checkpoint.pt")
+
+            net_using_post_localSGD_opt = torch.nn.parallel.DistributedDataParallel(
+                copy.deepcopy(DDP_NET).cuda(),
+                device_ids=[self.rank]
+            )
+            
+            # Process group cannot be pickled in some environments,
+            # so cannot deep copy an averager. See:
+            # https://github.com/pytorch/pytorch/pull/74737#pullrequestreview-922487496
+            averager = create_averager()
+            post_localSGD_opt = post_localSGD_optimizer.PostLocalSGDOptimizer(
+                optim=torch.optim.SGD(net_using_post_localSGD_opt.parameters(), lr=learning_rate),
+                averager=averager,
+            )
+            averager2 = create_averager()
+            dummy_post_localSGD_opt = post_localSGD_optimizer.PostLocalSGDOptimizer(
+                optim=torch.optim.SGD(net_using_post_localSGD_opt.parameters(), lr=learning_rate),
+                averager=averager2,
+            )
+
+            input = torch.randn(dist.get_world_size() * 2, 2).cuda()
+            target = torch.randn(dist.get_world_size() * 2, 4).cuda()
+            loss_fn = nn.MSELoss()
+
+            for _ in range(20):
+                
```

Comment:
See https://github.com/pytorch/pytorch/blob/master/CONTRIBUTING.md for instructions on linting.